File: service/src/main/java/io/camunda/service/security/auth/Authentication.java
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */
package io.camunda.service.security.auth;

import static io.camunda.util.CollectionUtil.addValuesToList;
import static io.camunda.util.CollectionUtil.addValuesToSet;

import io.camunda.service.search.filter.FilterBase;
import io.camunda.util.ObjectBuilder;
import java.util.Collection;
import java.util.List;
import java.util.Set;

public record Authentication(
    String authenticatedUserId,
    List<String> authenticatedGroupIds,
    Set<String> authenticatedTenantIds,
    String token)
    implements FilterBase {

  public static final class Builder implements ObjectBuilder<Authentication> {

    private String user;
    private List<String> groups;
    private Set<String> tenants;
    private String token;

    public Builder user(final String value) {
      user = value;
      return this;
    }

    public Builder group(final String value) {
      return groups(List.of(value));
    }

    public Builder groups(final List<String> values) {
      groups = addValuesToList(groups, values);
      return this;
    }

    public Builder tenant(final String tenant) {
      return tenants(Set.of(tenant));
    }

    public Builder tenants(final Collection<String> values) {
      tenants = addValuesToSet(tenants, values);
      return this;
    }

    public Builder token(final String value) {
      token = value;
      return this;
    }

    @Override
    public Authentication build() {
      return new Authentication(user, groups, tenants, token);
    }
  }
}

File: webapp/client/packages/camunda-api-zod-schemas/lib/8.8/authentication.ts
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

import {z} from 'zod';
import {API_VERSION, type Endpoint} from '../common';

const currentUserSchema = z.object({
	username: z.string(),
	displayName: z.string(),
	email: z.string(),
	authorizedComponents: z.array(z.string()),
	tenants: z.array(
		z.object({
			key: z.number(),
			tenantId: z.string(),
			name: z.string(),
			description: z.string().optional(),
		}),
	),
	groups: z.array(z.string()),
	roles: z.array(z.string()),
	salesPlanType: z.string().nullable(),
	c8Links: z.array(
		z.object({
			name: z.string(),
			link: z.string(),
		}),
	),
	canLogout: z.boolean(),
	apiUser: z.boolean().optional(),
});

type CurrentUser = z.infer<typeof currentUserSchema>;

const getCurrentUser: Endpoint = {
	method: 'GET',
	getUrl: () => `/${API_VERSION}/authentication/me`,
};

export {currentUserSchema, getCurrentUser};
export type {CurrentUser};
